### ISSUE TITLE
Custom spectrogram filterbanks

### ISSUE THREAD

**Issue by dvisockas** [pytorch/audio#943]
## 🚀 Feature
It would be great to have an ability to create non-mel and not linear filterbanking. 

## Motivation
Even though mel-scale ceptstral coefficients are (probably) the most popular, there exists many other ways to filterbank the frequencies of spectra - CQCC, Opus filterbanks, BFCC, etc. I was thinking that it would be nice if the `Spectrogram` transform would accept a filterbank argument and filter the spectrogram accordingly.

## Pitch
Extension of `Spectrogram` API to accept `filter_bank` argument (with a fallback to current implementation). Maybe it could be a tensor of shape (num_banks, filter) where filter would consist of (min_freq, max_freq) with a triangular filterbank for the start?

## Additional context

Somewhat relates to #942

**Comment by vincentqb** [comment]
Thanks for the suggestion :) This relates to [comment](https://github.com/pytorch/audio/pull/593#discussion_r417068529). Thoughts? Would #593 meet your needs?

**Comment by dvisockas** [comment]
#593 is related to mel filter_banks as `create_fb_matrix` calculates mel filterbanks. I was thinking that it would be nice to somehow have an API that would enable users to create non_mel spectrograms. It is a simple operation of `spectrogram * filter_bank`, but implementing non-mel filterbanks is rather tedious because one has to recreate `F.create_fb_matrix` themselves

**Comment by vincentqb** [comment]
Yes, that would be useful, thanks for suggesting :) How about something similar to [AmplitudeToDB](https://github.com/pytorch/audio/blob/master/torchaudio/transforms.py#L173) then?
